tony-core/src/main/java/com/linkedin/tony/ApplicationMaster.java

```diff
@@ -385,7 +385,10 @@ private boolean prepare() throws IOException {
 
     // Setup application RPC server
     String amHostname = Utils.getCurrentHostName();
-    applicationRpcServer = setupRPCService(amHostname);
+    ServerSocket amRpcSocket = new ServerSocket(0);
+    int amRpcPort = amRpcSocket.getLocalPort();
+    amRpcSocket.close();
```

Comment:
move this setupRPCService() ?

Reply:
@oliverhu  ServerSocket may throw IOException https://docs.oracle.com/javase/8/docs/api/java/net/ServerSocket.html
So, we have to handle it cautiously. If we move this code to setupRPCService() then 
we need to handle throw exception there. Is it OK?

Reply:
@oliverhu  As discussed, I moved the port allocation code inside the ApplicationRpcServer constructor.